### PR TITLE
C128 function ram emd

### PR DESCRIPTION
**Pull request by blackystardust**
Added c128 internal/external function ram emd's and added items to the em-test.c struct.

### CODE REVIEW

File: libsrc/c128/emd/c128-efnram.s [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+; ------------------------------------------------------------------------
+; INSTALL routine. Is called after the driver is loaded into memory. If
+; possible, check if the hardware is present and determine the amount of
+; memory available.
+; Must return an EM_ERR_xx code in a/x.
+;
+
+INSTALL:
+        sei
```
Comment (greg-king5) [inline libsrc/c128/emd/c128-efnram.s]: Interrupts should be disabled as late as possible; and, re-enabled as early as possible (i.e., they should be blocked for as short a time as possible).

File: libsrc/c128/emd/c128-efnram.s [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        ldx     #0
+        stx     ptr1
+        ldx     #$80
+        stx     ptr1+1
+        ldx     #<ptr1
+        stx     FETVEC
+        stx     STAVEC
+        ldy     #0
+        ldx     #MMU_CFG_EXT_FROM
+        jsr     FETCH
```
Comment (greg-king5) [inline libsrc/c128/emd/c128-efnram.s]: Just before that `jsr FETCH`
seems like a good place to put the `sei`.

File: libsrc/c128/emd/c128-efnram.s [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+        tax
+        inx
+        txa
+        sta     tmp1
+        ldx     #MMU_CFG_EXT_FROM
+        jsr     STASH
+        ldx     #MMU_CFG_EXT_FROM
+        jsr     FETCH
```
Comment (greg-king5) [inline libsrc/c128/emd/c128-efnram.s]: Just after that `jsr FETCH`
seems like a good place to put the `cli`.

File: libsrc/c128/emd/c128-efnram.s (hunk 1 above)
Comment (blackystardust) [inline libsrc/c128/emd/c128-efnram.s]: In that case a lot of other emd code will need to be 'fixed' as well.
Is the emphasis of an emd on code size, execution time or interrupt handling availability ??

File: asminc/c128.inc [hunk 4]
```diff
@@ -198,12 +198,14 @@ CIA2_CRB        := $DD0F
...
-MMU_CFG_RAM1    := %01111111    ; Bank 1 full RAM
-MMU_CFG_RAM2    := %10111111    ; Bank 2 full RAM
-MMU_CFG_RAM3    := %11111111    ; Bank 3 full RAM
+MMU_CR           := $FF00
+MMU_CFG_CC65     := %00001110    ; Bank 0 with kernal ROM
+MMU_CFG_RAM0     := %00111111    ; Bank 0 full RAM
+MMU_CFG_RAM1     := %01111111    ; Bank 1 full RAM
+MMU_CFG_RAM2     := %10111111    ; Bank 2 full RAM
+MMU_CFG_RAM3     := %11111111    ; Bank 3 full RAM
+MMU_CFG_INT_FROM := %01010111    ; Bank 1 with Internal Function RAM/ROM
```
Comment (greg-king5) [inline asminc/c128.inc]: MMU_CFG_IFROM   := %01010111    ; Bank 1 with Internal Function RAM/ROM

File: asminc/c128.inc [hunk 5]
```diff
@@ -198,12 +198,14 @@ CIA2_CRB        := $DD0F
...
+MMU_CFG_EXT_FROM := %01101011    ; Bank 1 with External Function RAM/ROM
```
Comment (greg-king5) [inline asminc/c128.inc]: We prefer that the Assembly fields are aligned on eight-character boundaries.  Therefore, I think that those last two names should be shortenned.  I suggest:

+ `MMU_CFG_IFROM`
+ `MMU_CFG_EFROM`

File: libsrc/c128/emd/c128-efnram.s (hunk 1 above)
Comment (oliverschmidt) [inline libsrc/c128/emd/c128-efnram.s]: Maybe I'm missing the point but at least in my cc65 user expereince I run much more often in memory shortage issues than performance issues. From that perspective I'd say that in general code size has the highest prio. However here are of course hot spots rather optimized for performance. An example coming into my mind are unrolled loops in CONIO `clrscr()` implementations.

**Pull request merged**